vm/package/execute.rs

```diff
@@ -0,0 +1,342 @@
+                }
+            }
+        }
+
+        // Prepare the build directory.
+        let build_directory = self.build_directory();
+        // Load the prover.
+        let prover = ProverFile::open(&build_directory, &function_name)?;
+        // Load the verifier.
+        let verifier = VerifierFile::open(&build_directory, &function_name)?;
+
+        // Adds the proving key to the process.
+        process.insert_proving_key(program_id, &function_name, prover.proving_key().clone())?;
+        // Adds the verifying key to the process.
+        process.insert_verifying_key(program_id, &function_name, verifier.verifying_key().clone())?;
+
+        // Execute the circuit.
+        let (execution_response, execution) = process.execute::<A, R>(authorization, rng)?;
+
+        // Execute the additional fee.
+        let (_, additional_fee) = process.execute_additional_fee::<A, _>(private_key, credits, 1, rng)?;
+
+        // TODO: the additional_fee should not be optional.
+        let execution_transaction = Transaction::from_execution(execution, Some(additional_fee))?;
+
+        let caller = self.manifest_file().development_address();
+
+        match endpoint {
+            Some(ref endpoint) => {
+                // Construct the deploy request.
+                let request = ExecutionRequest::new(execution_transaction, *caller, *program_id)?;
+                // Send the deploy request.
+                let response = request.send(endpoint)?;
+                Ok((execution_response, response.transaction))
+            }
+            None => Ok((execution_response, execution_transaction)),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    type CurrentNetwork = snarkvm_console::network::Testnet3;
+    type CurrentAleo = snarkvm_circuit::network::AleoV0;
+
+    #[test]
+    fn test_execute() {
+        // Samples a new package at a temporary directory.
+        let (directory, package) = crate::package::test_helpers::sample_package();
+
+        // Initialize caller.
+        let caller_private_key = package.manifest_file().development_private_key();
+        let caller = Address::try_from(caller_private_key).unwrap();
+        let caller_credits = Record::<CurrentNetwork, Plaintext<CurrentNetwork>>::from_str(&format!(
+            "{{ owner: {caller}.private, gates: 5u64.private, _nonce: 0group.public }}"
+        ))
+        .unwrap();
+
+        // Initialize the function inputs.
+        let r0 = Value::<CurrentNetwork>::from_str(&caller.to_string()).unwrap();
+        let r1 = Value::<CurrentNetwork>::from_str("99u64").unwrap();
+
+        // Deploy the package.
+        let (execution_response, execution_transaction) = package
+            .execute::<CurrentAleo, _>(
+                None,
+                caller_private_key,
+                caller_credits,
+                Identifier::from_str("mint").unwrap(),
+                &[r0, r1],
+                &mut rand::thread_rng(),
```

Comment:
`snarkvm_utilities::TestRng` should be used here instead (it registers its seed, so it's easy to reproduce if anything goes wrong)